src/Dictionaries/HashedDictionaryCollectionTraits.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <type_traits>
+#include <sparsehash/sparse_hash_map>
+#include <Common/HashTable/Hash.h>
+
+namespace DB
+{
+
+/// sparse_hash_map/sparse_hash_set
+template <typename C>
+concept IsSparseHashTable = std::is_same_v<C, google::sparse_hash_map<
```

Comment:
Looks to me that the term "SparseHashTable" is too loaded already. Maybe `IsGoogleSparseHashTable`?